internal/cloudfront/aws_models.go

```diff
@@ -134,10 +137,25 @@ func newAWSOrigin(o Origin) *cloudfront.Origin {
 				Items:    SSLProtocols,
 				Quantity: aws.Int64(int64(len(SSLProtocols))),
 			},
-		},
-		DomainName: aws.String(o.Host),
-		Id:         aws.String(o.Host),
-		OriginPath: aws.String(""),
+		}
+		originAccessControlID = nil
+		s3OriginConfig = nil
+	} else {
+		customOriginConfig = nil
+		originAccessControlID = &o.OAC.ID
+		s3OriginConfig = &cloudfront.S3OriginConfig{
+			OriginAccessIdentity: aws.String(""),
+		}
```

Comment:
Just for curiosity, does the AWS API require this field declared empty? LOL hehehe

Reply:
Ok, If is possible let's test with and without this field, wdyt?

Reply:
Created https://gympass.atlassian.net/browse/PE1-1298 for this specific test, to free up this PR.